widget/embedded/src/hooks/useWalletList.ts

```diff
@@ -97,9 +97,7 @@ export function useWalletList(params: Params) {
   };
 
   useEffect(() => {
-    return () => {
-      disconnectConnectingWallets();
-    };
+    disconnectConnectingWallets();
   }, []);
 
```

Comment:
I don't think it's a good idea to tidy up the state right after the component mounts for the first time or manage cleanup in another component using this hook (like the wallet page and the confirm wallets modal). It just spreads the logic all over.
You could go for something like this instead :
```ts
//utils.ts
export function hashWalletsState(walletsInfo: WalletInfo[]): string {
    return walletsInfo.map((w) => w.state).join('-');
  }
 
//useWalletList.ts

const disconnectConnectingWallets = useCallback(() => {
    const connectingWallets =
      wallets?.filter((wallet) => wallet.state === WalletState.CONNECTING) ||
      [];
    for (const wallet of connectingWallets) {
      void disconnect(wallet.type);
    }
  }, [hashWalletsState(wallets)]);

  useEffect(() => {
    return () => {
      disconnectConnectingWallets();
    };
  }, [disconnectConnectingWallets]);
```

Reply:
Fixed